docs/tutorials/csharp/basic_csharp.md

```diff
@@ -10,64 +10,59 @@ nav_order: 1
 
 # C# Tutorial: Basic
 
-Here is a simple tutorial for getting started with running inference on an existing ONNX model for a given input data.
-The model is typically trained using any of the well-known training frameworks and then exported into the ONNX format. 
+Learn how to get started with inference with the C# API. Run an ONNX model for a given input data. The model is typically trained using any of the well-known training frameworks and then exported into the ONNX format.
 
-Note, that the following classes `NamedOnnxValue`, `DisposableNamedOnnxValue`, `FixedBufferOnnxValue` are going
-to be deprecated in the future. They are not recommended for new code.
+## OrtValue API
 
-The new `OrtValue` based API is the recommended approach. The `OrtValue` API generates less garbage and is more performant.
-Some scenarios indicated 4x performance improvement over the previous API and significantly less garbage.
-It provides uniform access to data via `ReadOnlySpan<T>` and `Span<T>` structures, regardless of its location, managed or unmanaged.
+The new `OrtValue` based API is the recommended approach. The `OrtValue` API generates less garbage and is more performant. Some scenarios show 4x performance improvement over the previous API and significantly less garbage.
 
-`DenseTensor` class can still be used for multi-dimensional access to the data since the new `Span` based API feature
-only a 1-D index. However, some reported a slow performance when using `DenseTensor` class multi-dimensional access.
-One can then create an OrtValue on top of the tensors data. 
+OrtValue is a universal container that can hold different ONNX types, such as tensors, maps, and sequences. It always existed in the onnxruntime library, but was not exposed in the C# API.
+
+The `OrtValue` based API provides uniform access to data via `ReadOnlySpan<T>` and `Span<T>` structures, regardless of its location, managed or unmanaged.
+
+Note, that the following classes `NamedOnnxValue`, `DisposableNamedOnnxValue`, `FixedBufferOnnxValue` will be deprecated in the future. They are not recommended for new code.
+
+## Data shape
+
+`DenseTensor` class can be used for multi-dimensional access to the data since the new `Span` based API features only a 1-D index. However, some reported a slow performance when using `DenseTensor` class multi-dimensional access. One can then create an OrtValue on top of the tensors data. 
```

Comment:
```suggestion
`DenseTensor` class can be used for multi-dimensional access to the data since the new `Span` based API features only a 1-D index. However, some reported a slow performance when using `DenseTensor` class multi-dimensional access. One can then create an OrtValue on top of the DenseTensor
```suggestion
`DenseTensor` class can be used for multi-dimensional access to the data since the new `Span` based API features only a 1-D index. However, some reported a slow performance when using `DenseTensor` class multi-dimensional access. One can then create an OrtValue on top of the tensors data. 
``` data. 
```